spec/core/ics-004-channel-and-packet-semantics/UPGRADES.md

```diff
@@ -855,12 +885,12 @@ function timeoutChannelUpgrade(
 }
 ```
 
-Note that the timeout logic only applies to the INIT step. This is to protect an upgrading chain from being stuck in a non-OPEN state if the counterparty cannot execute the TRY successfully. Once the TRY step succeeds, then both sides are guaranteed to have the upgrade feature enabled. Liveness is no longer an issue, because we can wait until liveness is restored to execute the ACK step which will move the channel definitely into an OPEN state (either a successful upgrade or a rollback).
+Both parties must not complete the upgrade handshake if the counterparty upgrade timeout has already passed. Even if both sides could have successfully moved to FLUSHCOMPLETE. This will prevent the channel ends from reaching incompatible states.
```

Comment:
minor nit: if the timeout has passed, but each chain moved to flush complete before the timeout passed. That's fine